### PR TITLE
[SYCL] Add a test for freeing USM before kernel execution is finished

### PR DESCRIPTION
**Pull request by sergey-semenov**
Tests changes from https://github.com/intel/llvm/pull/4928

### CODE REVIEW

File: SYCL/USM/free_during_kernel_execution.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        auto Accessor = Buffer.get_access<sycl::access::mode::write>(cgh);
+        cgh.parallel_for<KernelA>(NumOfWorkItems, [=](sycl::id<1> WIid)
+        {
+            Accessor[WIid] = WIid.get(0);
+        });
+    });
+
+    // Check that freeing USM before the kernel was finished works.
+    free(USM, Queue.get_context());
```
Comment (s-kanaev) [inline SYCL/USM/free_during_kernel_execution.cpp]: How is the check performed?
Should one use cl or ze intercept?

Comment (sergey-semenov) [reply SYCL/USM/free_during_kernel_execution.cpp]: Discussed this with @s-kanaev offline, we currently don't have unit tests for OpenCL plugin in order to check what `piextUSMFree` is lowered to, so we agreed on checking that free works (i. e. does not crash) as an E2E test.

**Review by s-kanaev — APPROVED** [review]
LGTM

**Comment by sergey-semenov** [comment]
@jbrodman Could you please take a look?

**Review by jbrodman — APPROVED** [review]
LGTM

**Pull request merged**